script/sign-json.rb

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env ruby
+require "base64"
+require "json"
+require "openssl"
+require "pathname"
+
+ROOT = Pathname(__dir__).parent.freeze
+KEY_ID = ENV.fetch("JWS_SIGNING_KEY_ID").freeze
+PRIVATE_KEY = OpenSSL::PKey::RSA.new(ENV.fetch("JWS_SIGNING_KEY")).freeze
+
+[
+  ROOT/"_site/api/formula.json",
+  ROOT/"_site/api/cask.json",
+].each do |path|
+  data_string = path.read
+
+  # References:
+  # - RFC7515 [JSON Web Signature (JWS)]
+  # - RFC7797 [JWS Unencoded Payload Option]
+
+  header = {
+    "alg": "PS512",
+    "b64": false,
+    "crit": ["b64"],
```

Comment:
Explaining why each of these values is what it is today would be handy. Maybe worth turning `"b64"` into a variable, too, as it seems to be shared here.

Reply:
Added (probably too much) comments.